reactor-core/src/main/java/reactor/core/publisher/FluxSubscribeOnValue.java

```diff
@@ -218,20 +225,30 @@ public void clear() {
 		}
 	}
 
-	static final class ScheduledEmpty implements QueueSubscription<Void>, Runnable {
+	static final class ScheduledEmpty implements QueueSubscription<Void>,
+	                                             Scheduler.ContextRunnable {
 
 		final Subscriber<?> actual;
 
+		final Context context;
```

Comment:
could this be replaced by just using the `actual.currentContext()` ? the only usage of `ScheduledEmpty` is with `CoreSubscriber` actuals, so you could refine the type of the field and constructor parameter.